cmake/projects/OpenSSL/schemes/url_sha1_openssl_macos.cmake.in

```diff
@@ -0,0 +1,216 @@
+  hunter_status_debug("Using default macOS architecture: x86_64")
+  set(configure_architectures "x86_64")
+endif()
+
+foreach(arch ${configure_architectures})
+  list(APPEND ssl_input_libraries "@HUNTER_PACKAGE_SOURCE_DIR@/${arch}/libssl.a")
+  list(APPEND crypto_input_libraries "@HUNTER_PACKAGE_SOURCE_DIR@/${arch}/libcrypto.a")
+
+  if(BUILD_SHARED_LIBS)
+    list(APPEND ssl_input_shared_libraries "@HUNTER_PACKAGE_SOURCE_DIR@/${arch}/libssl.dylib")
+    list(APPEND crypto_input_shared_libraries "@HUNTER_PACKAGE_SOURCE_DIR@/${arch}/libcrypto.dylib")
+  endif()
+endforeach()
+
+# Skip position independent code for macOS
+set(skip_pic SKIP_PIC)
+set(skip_includes SKIP_INCLUDES)
+hunter_dump_cmake_flags(${skip_includes} ${skip_pic})
+
+# Pass C compiler through
+set(configure_command
+    MACHINE=${CMAKE_SYSTEM_PROCESSOR}
+    CC=${CMAKE_C_COMPILER}
+    ${configure_command})
+
+# Pass C flags through
+set(configure_opts ${CMAKE_C_FLAGS})
+
+# Add clear-all to commands
+set(configure_command . "@HUNTER_GLOBAL_SCRIPT_DIR@/clear-all.sh" && "${configure_command}")
+set(build_command . "@HUNTER_GLOBAL_SCRIPT_DIR@/clear-all.sh" && make)
+
+# Configuration options
+if(BUILD_SHARED_LIBS)
+  set(shared_flag shared)
+else()
+  set(shared_flag no-shared)
+endif()
+set(configure_opts ${configure_opts} threads ${shared_flag})
+
+hunter_status_debug("Configure command: ${configure_command}")
+hunter_status_debug("Configure architectures: ${configure_architectures}")
+hunter_status_debug("Configure options: ${configure_opts}")
+
+
+# fuse all libraries with 'lipo'
+ExternalProject_Add(
+    "@HUNTER_EP_NAME@-macos_universal"
+    DOWNLOAD_COMMAND
+    ""
+    DOWNLOAD_DIR
+    "@HUNTER_PACKAGE_DOWNLOAD_DIR@"
+    TLS_VERIFY
+    "@HUNTER_TLS_VERIFY@"
+    SOURCE_DIR
+    "@HUNTER_PACKAGE_SOURCE_DIR@/macos_universal"
+    INSTALL_DIR
+    "@HUNTER_PACKAGE_INSTALL_PREFIX@"
+        # not used, just avoid creating Install/<name> empty directory
+    CONFIGURE_COMMAND
+    ""
+    BUILD_COMMAND
+    ""
+    INSTALL_COMMAND
+    lipo
+    -create
+    ${ssl_input_libraries}
+    -o
+    "@HUNTER_PACKAGE_INSTALL_PREFIX@/lib/libssl.a"
+    COMMAND
+    lipo
+    -create
+    ${crypto_input_libraries}
+    -o
+    "@HUNTER_PACKAGE_INSTALL_PREFIX@/lib/libcrypto.a"
+)
+
+if(BUILD_SHARED_LIBS)
+  # fuse shared libraries with 'lipo'
+  ExternalProject_Add(
+      "@HUNTER_EP_NAME@-macos_universal_shared"
+      DOWNLOAD_COMMAND
+      ""
+      DOWNLOAD_DIR
+      "@HUNTER_PACKAGE_DOWNLOAD_DIR@"
+      TLS_VERIFY
+      "@HUNTER_TLS_VERIFY@"
+      SOURCE_DIR
+      "@HUNTER_PACKAGE_SOURCE_DIR@/macos_universal"
+      INSTALL_DIR
+      "@HUNTER_PACKAGE_INSTALL_PREFIX@"
+          # not used, just avoid creating Install/<name> empty directory
+      CONFIGURE_COMMAND
+      ""
+      BUILD_COMMAND
+      ""
+      INSTALL_COMMAND
+      lipo
+      -create
+      ${ssl_input_shared_libraries}
+      -o
+      "@HUNTER_PACKAGE_INSTALL_PREFIX@/lib/libssl.dylib"
```

Comment:
CMake will build shared library if `BUILD_SHARED_LIBS=ON` (without static library) and static library if `BUILD_SHARED_LIBS=OFF` (without shared library). I think here we should behave in the same way. Hence logic can be simplified like this:
```cmake
if(BUILD_SHARED_LIBS)
  set(libname libcrypto.dylib)
else()
  set(libname libcrypto.a)
endif()
```

Reply:
Actually, OpenSSL will build both static and shared libraries when the `shared` flag is passed; In this case the static library will load the shared library. 

See here: https://wiki.openssl.org/index.php/Compilation_and_Installation#Configure_Options

>Note: when building a shared object, both the static archive and shared objects are built. You do not need to do anything special to build both when shared is specified.

Reply:
CMake users will not have access to second library, hence it still means that you can have something like
```cmake
if(BUILD_SHARED_LIBS)
  set(libname libcrypto.dylib)
else()
  set(libname libcrypto.a)
endif()
```

Build instructions can be simplified, I think there is no need to have second companioning `ExternalProject_Add` command.